Zend/zend_inheritance.c

```diff
@@ -359,6 +362,9 @@ static zend_bool zend_do_perform_implementation_check(const zend_function *fe, c
 	if (proto->common.fn_flags & ZEND_ACC_HAS_RETURN_TYPE) {
 		/* Removing a return type is not valid. */
 		if (!(fe->common.fn_flags & ZEND_ACC_HAS_RETURN_TYPE)) {
+			if (ZEND_TYPE_CODE(proto->common.arg_info[-1].type) == IS_MIXED) {
+				return 1;
+			}
 			return 0;
```

Comment:
I believe this can be simplified:
```suggestion
			return ZEND_TYPE_CODE(proto->common.arg_info[-1].type) == IS_MIXED;
```